config.go

```diff
@@ -58,7 +58,7 @@ type DatabaseConfig struct {
 	TLS *TLSConfig
 }
 
-func (c *DatabaseConfig) MySQLConfig() (*mysql.Config, error) {
+func (c DatabaseConfig) MySQLConfig() (*mysql.Config, error) {
```

Comment:
Why was this reverted? Since some methods are defined as pointer receivers, all methods on this struct should be pointer receivers. From the [golang FAQ](https://golang.org/doc/faq#methods_on_values_or_pointers)

> Next is consistency. If some of the methods of the type must have pointer receivers, the rest should too, so the method set is consistent regardless of how the type is used. See the section on method sets for details.